sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -563,4 +572,40 @@ export class StudentService extends RecordDataModelService<Student> {
 
   @InjectLogger()
   logger: LoggerService;
+
+  /**
+   * Search for the student's SFAS data and update overaward values in
+   * disbursement overawards table if any overaward values for BCSL or CSLF exist.
+   * @param student student object.
+   * @param sinNumber student's sin number.
+   * @param entityManager entityManager to be used to perform the query.
+   */
+  async updateSFASOveraward(
+    student: Student,
+    sinNumber: string,
+    entityManager: EntityManager,
+  ) {
+    const sfasIndividual = await this.sfasIndividualService.getSFASOverawards(
+      student.user.lastName,
+      student.birthDate,
+      sinNumber,
+    );
+
+    if (sfasIndividual?.bcslOveraward > 0) {
```

Comment:
May be not an issue here because it will always be 0 to 2 records to be added, but just keep in mind that the objects records can be created and provided to Typeorm as an array also 😉

Reply:
Sorry but I didn't get what you mean by "objects records can be created and provided to Typeorm as an array".

Reply:
For instance:
```ts
await applicationRepository.insert([application, newApplication]);
```
or
```ts
await applicationRepository.save([application, newApplication]);
```

Reply:
I didn't understand the context

Reply:
I agree @andrewsignori-aot

Reply:
ok. Saving the array now.